### DISCUSSION TITLE
Modify the app `build.gradle` `defaultConfig` section

### DISCUSSION THREAD

**Discussion by gounthar** [dipien/semantic-version-gradle-plugin#17] [General]
Hello there 👋 

First of all, thanks for your work. Now, without further ado, let's get to the point.

**Is your feature request related to a problem? Please describe.**
I have a very simple Android application, and I'd like to be able to increment the `versionCode` and `versionName` which are described within the app `build.gradle` file.
The `printVersion` command gives me the output I'm looking for:

```bash
Version: 1.0.1
Version code: 100
Version name: 1.0.1
```
Calling the `incrementVersion` changes the value of `version` in the main `build.gradle`, but not in the `app/build.gradle` in the `defaultConfig` section.

**Describe the solution you'd like**
I would like the `versionCode` and `versionName` to be incremented in the `defaultConfig` section of the `app/build.gradle` file.

**Describe alternatives you've considered**

**Additional context**
Thank you.

**Comment by maxirosson** [comment]
Hi. We already support that. Are you using our android plugin?

`apply plugin: "com.dipien.android.semantic-version"`

**Comment by gounthar** [comment]
Thanks for your answer.
Yes [indeed](https://github.com/gounthar/MyFirstAndroidAppBuiltByJenkins/blob/da10503a9f7fd47ff1adc56886ca4a51b3406e26/build.gradle#L17). 🤔

**Comment by maxirosson** [comment]
The plugin automatically overrides the version name and version code on build time. You shouldn't include those fields on the defaultConfig section.
Thanks for reporting this. Given this is not clear, I am going to include this in the documentation and also change the plugin to fail if the project is explicitly defining those fields.

**Comment by gounthar** [comment]
Great, I'm going to implement these changes and let you know.

**Comment by gounthar** [comment]
Works for me now, thanks for taking the time to explain. 👍

**Comment by gounthar** [comment]
I'm sorry, but I still have questions...
Why do I still have this message?
```bash
> Configure project :
Version not specified on project [My First Built by Jenkins Applications] or its root project. Assigned v0.1.0 as default version
```
I have the version defined in the main `build.gradle`: `version = "1.0.4-ALPHA"`.

There is something else I haven't understood yet. I always get

```bash
Version: 1.0.4-ALPHA
```
when I use `./gradlew printVersion -Palpha=false` or `./gradlew printVersion -Palpha=true`.

Thank you.

**Comment by gounthar** [comment]
In the documentation I read:

> The Semantic Version Gradle Plugin automatically generates and configures the versionCode and versionName on the defaultConfig section at build time. It uses the version defined at the root build.gradle[.kts] to configure those fields. It's important to remember you don't have to set those fields on the defaultConfig section.

Do we agree that I should not add a `versionCode` anywhere in the Gradle configuration or in the `AndroidManifest.xml` ?

I have then a problem when I try to publish my new version on Google Play Console, as I get:
```bash
Version code is too low or has already been used for app io.jenkins.mobile.example.myfirstbuiltbyjenkinsapplication
```

It looks like when I use the ` incrementVersion --versionIncrementType=PATCH` task, the `versionCode` does not get incremented.
I have added a `android:versionCode="101"` in the `AndroidManifest.xml` file, but that hasn't changed the behaviour.

Thanks.

**Comment by maxirosson** [comment]
I think both problems are related to the same issue. You should declare the version before applying the plugin. I am going to clarify that on the documentation

**Reply by gounthar** [reply]
It looks like I'm making every mistake than can be made, sorry about that.

Thanks, it now works 👍 .

**Comment by maxirosson** [comment]
No, thanks for that. I could improve the documentation and plugin validations thanks to your comments. I am releasing a new version that fails if the version is not assigned before applying the plugin
